eox_core/api/v1/serializers.py

```diff
@@ -79,6 +91,77 @@ def validate(self, attrs):
         return attrs
 
 
+class EdxappExtendedUserSerializer(EdxappUserSerializer):
+    """
+    This serializer allows admiting all the
+    UserProfile fields and also the extra fields for the
+    userProfile.meta values that are specified in the
+    microsite settings.
+    """
+    year_of_birth = serializers.ChoiceField(choices=YEAR_OF_BIRTH_CHOICES)
+    gender = serializers.ChoiceField(choices=get_gender_choices())
+    city = serializers.CharField()
+    goals = serializers.CharField()
+    bio = serializers.CharField(max_length=3000)
+    profile_image_uploaded_at = serializers.DateTimeField()
+    phone_number = serializers.CharField(max_length=50)
+    mailing_address = serializers.CharField()
+    courseware = serializers.CharField(max_length=255)
+    level_of_education = serializers.ChoiceField(choices=get_level_of_education_choices())
+    country = CountryField(required=False)
+
+    def __init__(self, *args, **kwargs):  # pylint: disable=too-many-locals
+        """
+        Add the custom registration fields specified in the settings.
+        """
+        super().__init__(*args, **kwargs)
+
+        extended_profile_fields = getattr(settings, 'extended_profile_fields', [])
+        extra_fields = getattr(settings, 'REGISTRATION_EXTRA_FIELDS', {})
+        ednx_custom_registration_fields = getattr(settings, 'EDNX_CUSTOM_REGISTRATION_FIELDS', [])
+        all_fields = set(self.fields)
+        # Pop irrelevant fields for the API
+        for field in ["honor_code", "terms_of_service"]:
+            extra_fields.pop(field, None)
+        # Obtain only the fields defined in the EdxappExtendedUserSerializer
+        non_profile_fields = set(EdxappUserSerializer().fields)
+        for field in ["activate_user", "skip_password"]:
+            non_profile_fields.add(field)
+        profile_fields = all_fields - non_profile_fields
+
+        # Delete the profile fields that are not allowed or are redefined in the ednx_custom_registration setting
+        # In case the field IS allowed, check if is required or not
+        for field in profile_fields:
+            if field not in extra_fields or field in extended_profile_fields:
+                self.fields.pop(field)
+            else:
+                self.fields[field].required = extra_fields.get(field) == "required"
+
+        # Adding fields that go inside the UserProfile.meta
+        for custom_field in ednx_custom_registration_fields:
+            field_name = custom_field.get("name")
+            field_type = custom_field.get("type")
+
+            if field_name in extended_profile_fields and field_type in ALLOWED_TYPES:
+                serializer_field = {}
+
+                serializer_field["required"] = extra_fields.get(field_name) == "required"
+
+                # Check this first since fields of type 'text' are the only ones that can have restrictions
+                if field_type in ["text"]:
```

Comment:
```suggestion
                if field_type == "text":
```
Better to be explicit in the equality check

Reply:
@MoisesGSalas  Done